src/linear_samplers/vec_row_uniform_sparse.jl

```diff
@@ -0,0 +1,56 @@
+# This file is pat of RLinearAlgebra.jl
+# 1. Specifies type
+# 2. Implements sample function
+# 3. Exports Type
+
+# using Random
+
+"""
+    LinSysVecRowSparseUnifSampler <: LinSysVecRowSelect
+
+A mutable structure that specifies sampling a proporition of the
+rows of a linear system, scaling each using independent uniform 
+random variables, and then taking their sum. The proportion of rows
+sampled without replacement is given by (at most) `sparsity` which must be 
+between 0 and 1 (not inclusive).
+
+# Fields
+- `sparsity::Float64`
+
+# Constructors
+- `LinSysVecRowSparseUnifSampler()` defaults the `sparsity` level to 0.2.
+"""
+mutable struct LinSysVecRowSparseUnifSampler <: LinSysVecRowSelect
+    sparsity::Float64
+    function LinSysVecRowSparseUnifSampler(sparsity::Float64)
+        if sparsity <= 0.0 || sparsity >= 1.0
+            DomainError(sparsity, "Must be strictly between 0.0 and 1.0") |>
+                throw
+        else
+            return new(sparsity)
+        end
+    end
+end
+
+LinSysVecRowSparseUnifSampler() = LinSysVecRowSparseUnifSampler(0.2)
+
+function sample(
+    type::LinSysVecRowSparseUnifSampler,
+    A::AbstractArray,
+    b::AbstractVector,
+    x::AbstractVector,
+    iter::Int64
+)
+
+    # Determine indices of equations from the system to use
+    numrows = floor(Int64, type.sparsity * size(A,1))
```

Comment:
@vp314 what if sparsity == 0?

Reply:
`sparsity != 0` as this is checked for in the constructor. But, it is possible that the `numrows` can end up being zero. I pushed a fix that requires the `numrows` to be at least 1

Reply:
Addressed in updated PR.